core/templates/domain/exploration/read-only-exploration-backend-api.service.ts

```diff
@@ -17,152 +17,162 @@
  * about explorations from the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-
-require(
-  'pages/exploration-player-page/exploration-player-page.constants.ajs.ts');
-
-angular.module('oppia').factory('ReadOnlyExplorationBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'EXPLORATION_DATA_URL_TEMPLATE', 'EXPLORATION_VERSION_DATA_URL_TEMPLATE',
-  function($http, $q, UrlInterpolationService,
-      EXPLORATION_DATA_URL_TEMPLATE, EXPLORATION_VERSION_DATA_URL_TEMPLATE) {
-    // Maps previously loaded explorations to their IDs.
-    var _explorationCache = [];
-
-    var _fetchExploration = function(
-        explorationId, version, successCallback, errorCallback) {
-      var explorationDataUrl = _getExplorationUrl(explorationId, version);
-
-      $http.get(explorationDataUrl).then(function(response) {
-        var exploration = angular.copy(response.data);
-        if (successCallback) {
-          successCallback(exploration);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
-    };
-
-    var _isCached = function(explorationId) {
-      return _explorationCache.hasOwnProperty(explorationId);
-    };
-
-    var _getExplorationUrl = function(explorationId, version) {
-      if (version) {
-        return UrlInterpolationService.interpolateUrl(
-          EXPLORATION_VERSION_DATA_URL_TEMPLATE, {
-            exploration_id: explorationId,
-            version: String(version)
-          });
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { AppConstants } from 'app.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ReadOnlyExplorationBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService) {}
+  // Maps previously loaded explorations to their IDs.
+  private _explorationCache = [];
+
+  private _fetchExploration(
+      explorationId: string, version: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var explorationDataUrl = this._getExplorationUrl(explorationId, version);
+
+    this.http.get(explorationDataUrl).toPromise().then((response) => {
+      var exploration = cloneDeep(response);
```

Comment:
Use `let` here and in other places in this file.

Reply:
Done